Automattic/wp-e2e-tests
# Fix signup spec

We had some problems executing `wp-signup-spec.js` locally - if one test fails, it will affect every next test in line and they will fail also. After several different tries and approaches, I concluded that some tests were missing account deletion at the end, newly registered user is not signed out and that makes a problem for flow in the next test. 

This PR ensures that every new account is deleted, and instead of `step`, `deleteAccount()` is in `after` block. That way, if test fails account will be deleted. 

To test:
Run `wp-signup-spec.js` and make sure it passes. If some test fails, next test shouldn't fail.
